nodego/supervisor.go

```diff
@@ -0,0 +1,162 @@
+package nodego
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"runtime/debug"
+	"strconv"
+	"sync"
+	"time"
+
+	"golang.org/x/net/context"
+)
+
+type logEntry struct {
+	TextPayload string
+	Severity    string
+	Time        string
+	ExecutionID string
+}
+
+type logBatch struct {
+	Entries []*logEntry
+
+	payloadLength int
+	ready         chan struct{}
+}
+
+func (batch *logBatch) AddEntry(entry *logEntry) {
+	if batch.Entries == nil {
+		close(batch.ready)
+	}
+	batch.Entries = append(batch.Entries, entry)
+	batch.payloadLength += len(entry.TextPayload)
+}
+
+var (
+	currentLogBatch      *logBatch
+	currentLogBatchMutex sync.Mutex
+	logBatchesToReport   = make(chan *logBatch, 5)
+)
+
+var (
+	functionExecutionId string
+)
+
+var (
+	infoLogWriter  = &supervisorWriter{"INFO"}
+	errorLogWriter = &supervisorWriter{"ERROR"}
+
+	InfoLogger  = log.New(infoLogWriter, "", 0)
+	ErrorLogger = log.New(errorLogWriter, "", 0)
+)
+
+func init() {
+	startNewLogBatch()
+	go logBatchReportWorker()
+}
+
+func startNewLogBatch() *logBatch {
+	currentLogBatch = &logBatch{
+		ready: make(chan struct{}),
+	}
+	logBatchesToReport <- currentLogBatch
+	return currentLogBatch
+}
+
+type supervisorWriter struct {
+	severity string
+}
+
+func (w *supervisorWriter) Write(p []byte) (n int, err error) {
```

Comment:
It seems like we aught to be able to detect if we are running locally and log to STDERR.